File: MY_REPOS/Lambda-Resource-Static-Assets/2-resources/BLOG/Data-Structures/1-Python/binary-search-tree-solution-python/binary-search-tree-solution/leet_code_108.py
"""
View the full problem and run the test cases at:
https://leetcode.com/problems/convert-sorted-array-to-binary-search-tree/
"""


class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


def sortedArrayToBST(nums):
    """
  :type nums: List[int]
  :rtype: TreeNode
  """
    if len(nums) == 0:
        return None

    mid_idx = int(len(nums) / 2)
    root = TreeNode(nums[mid_idx])
    root.left = sortedArrayToBST(nums[:mid_idx])
    root.right = sortedArrayToBST(nums[mid_idx + 1 :])
    return root


bst_root = sortedArrayToBST([-10, -3, 0, 5, 9])
print(bst_root.val)  # 0
print(bst_root.left.val)  # -3
print(bst_root.left.left.val)  # -10
print(bst_root.right.val)  # 9
print(bst_root.right.left.val)  # 5
# Returns the root of the following binary search tree:
#       0
#      / \
#    -3   9
#    /   /
#  -10  5

File: MY_REPOS/Lambda-Resource-Static-Assets/2-resources/BLOG/ciriculumn/week-8/week8-projects/binaryTree/bst/starter/problems/leet_code_108.js
// View the full problem and run the test cases at:
//  https://leetcode.com/problems/convert-sorted-array-to-binary-search-tree/

function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

let nums = [-10,-3,0,5,9];
function sortedArrayToBST(nums) {
    for(let i = 0; i < nums.length; i++){
        nums[i]
        console.log("sortedArrayToBST -> nums[i]", nums[i])
        nums[i]
        console.log("sortedArrayToBST -> nums[i]", nums[i])
    }


}
sortedArrayToBST(nums)